File: backend/app/schemas/alerts.py
from datetime import datetime
from typing import Optional, List

from pydantic import BaseModel, ConfigDict


class AlertConfigUpdate(BaseModel):
    page_id: str
    telegram_enabled: bool = False
    telegram_chat_ids: List[str] = []
    email_enabled: bool = False
    email_recipients: List[str] = []
    dedup_window_minutes: int = 30


class AlertConfigResponse(BaseModel):
    page_id: str
    page_name: str = ""
    telegram_enabled: bool = False
    telegram_chat_ids: List[str] = []
    email_enabled: bool = False
    email_recipients: List[str] = []
    dedup_window_minutes: int = 30


class AlertLogEntry(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    type: str
    message: str
    severity: str
    channel: str
    page_name: Optional[str] = None
    sent_at: Optional[datetime] = None
    acknowledged: bool = False


class AlertLogResponse(BaseModel):
    entries: List[AlertLogEntry]
    total: int

File: frontend/src/api/alerts.ts
import apiClient from './client'

export interface AlertConfig {
  page_id: string
  page_name: string
  telegram_enabled: boolean
  telegram_chat_ids: string[]
  email_enabled: boolean
  email_recipients: string[]
  dedup_window_minutes: number
}

export interface AlertConfigUpdate {
  page_id: string
  telegram_enabled: boolean
  telegram_chat_ids: string[]
  email_enabled: boolean
  email_recipients: string[]
  dedup_window_minutes: number
}

export async function getAlertConfigs(): Promise<AlertConfig[]> {
  const { data } = await apiClient.get('/alerts/config')
  return data.configs
}

export async function updateAlertConfig(config: AlertConfigUpdate) {
  await apiClient.put('/alerts/config', config)
}
